scopes/component/component/ui/use-component-query.ts

```diff
@@ -186,9 +192,14 @@ export function useComponentQuery(componentId: string, host: string) {
   }, []);
 
   const rawComponent = data?.getHost?.get;
-
   return useMemo(() => {
+    const aspects = {
+      entries: rawComponent?.aspects?.map((aspect) => ({ aspectId: aspect.id, aspectData: aspect.data })),
```

Comment:
this functionality shouldn't be here. it should be part of the aspect-list or aspect-map
it should know to create it from the raw data (same as you did with the `ComponentDescriptor.fromObject` 2 lines below

Reply:
@odedre